https://bugs.gentoo.org/466924

From 31fc8a0b2f4eaf15155d5b02b8017220c32aacbd Mon Sep 17 00:00:00 2001
From: Nick Clifton <nickc@redhat.com>
Date: Thu, 14 Nov 2013 14:24:34 +0000
Subject: [PATCH] 	PR ld/16082 	* elf32-hppa.c
 (elf32_hppa_hide_symbol): Remove old version 	information when forcing a
 symbol to be local.

---
 bfd/ChangeLog    | 6 ++++++
 bfd/elf32-hppa.c | 4 ++++
 2 files changed, 10 insertions(+)

2013-11-14  Guy Martin <gmsoft@tuxicoman.be>

	PR ld/16082
	* elf32-hppa.c (elf32_hppa_hide_symbol): Remove old version
	information when forcing a symbol to be local.

diff --git a/bfd/elf32-hppa.c b/bfd/elf32-hppa.c
index 7d1725f..df25ac9 100644
--- a/bfd/elf32-hppa.c
+++ b/bfd/elf32-hppa.c
@@ -1771,6 +1771,10 @@ elf32_hppa_hide_symbol (struct bfd_link_info *info,
 	  _bfd_elf_strtab_delref (elf_hash_table (info)->dynstr,
 				  eh->dynstr_index);
 	}
+
+      /* PR 16082: Remove version information from hidden symbol.  */
+      eh->verinfo.verdef = NULL;
+      eh->verinfo.vertree = NULL;
     }
 
   /* STT_GNU_IFUNC symbol must go through PLT.  */
-- 
2.0.0

